src/routes/accounts.py

```
@@ -17,9 +28,269 @@
+
+    if stored_token.expires_at.replace(tzinfo=timezone.utc) < datetime.now(timezone.utc):
+        raise HTTPException(status_code=400, detail="Token has expired.")
+
+    user_result = await db.execute(select(UserModel).where(UserModel.id == user_id))
+    user = user_result.scalars().first()
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found.")
+
+    new_access_token = jwt_manager.create_access_token({"user_id": user.id})
+
+    return {"access_token": new_access_token}
+
+
+@router.post("/password-reset/request/")
+async def request_password_reset(
+        payload: PasswordResetRequestSchema,
+        db: AsyncSession = Depends(get_db)
+):
+    generic_response = {"message": "If you are registered, you will receive an email with instructions."}
+
+    result = await db.execute(select(UserModel).where(UserModel.email == payload.email))
+    user = result.scalars().first()
+
+    if not user or not user.is_active:
+        return generic_response
+
+    # Видаляємо старі токени
+    existing_tokens = await db.execute(
+        select(PasswordResetTokenModel).where(PasswordResetTokenModel.user_id == user.id)
+    )
+    for token in existing_tokens.scalars():
+        await db.delete(token)
+
+    token_str = str(uuid.uuid4())
+    reset_token = PasswordResetTokenModel(
+        user_id=user.id,
+        token=token_str,
+        expires_at=datetime.now(timezone.utc) + timedelta(hours=1)
+    )
+    db.add(reset_token)
+    await db.commit()
+
+    return generic_response
+
+
+@router.post("/reset-password/complete/")
+async def reset_password_complete(
+        payload: PasswordResetCompleteRequestSchema,
+        db: AsyncSession = Depends(get_db)
+):
+    try:
+        result = await db.execute(select(UserModel).where(UserModel.email == payload.email))
+        user = result.scalars().first()
+
+        if not user:
+            raise HTTPException(status_code=400, detail="Invalid email or token.")
```

According to the task description, this endpoint should return an error if the user is inactive. This conditional statement only checks for the user's existence. You should also add a check for `user.is_active` to prevent password resets on inactive accounts.